### PR TITLE
Refactoring: Move common code into velocitas_lib

### PR DESCRIPTION
**Pull request by doosuu**
This change moves common code into velocitas_lib for easier maintenance.

### CODE REVIEW

File: runtime-local/src/lib.py [hunk 1]
```diff
@@ -259,27 +230,20 @@ def stop_container(service_id, log=None):
...
-    log.write(f"Stopping {service_id!r}\n")
-
-    for config_entry in service_spec["config"]:
-        if config_entry["key"] == "no-dapr":
-            no_dapr = config_entry["value"]
+    log = create_log_file(service.id, "runtime-local")
+    log.write(f"Stopping {service.id!r}\n")
 
-    if not no_dapr and get_middleware_type() == MiddlewareType.DAPR:
+    if not service.config.no_dapr and get_middleware_type() == MiddlewareType.DAPR:
```
Comment (MP91) [inline runtime-local/src/lib.py]: The middleware definition and getter could also be moved to the velocitas lib

File: velocitas_lib/src/velocitas_lib/__init__.py [hunk 2]
```diff
@@ -68,34 +68,12 @@ def get_package_path() -> Path:
...
-    return json.load(
-        open(
-            path,
-            encoding="utf-8",
-        )
-    )
-
-
-def json_obj_to_flat_map(obj, prefix: str = "", separator: str = ".") -> dict[str, str]:
+def json_obj_to_flat_map(obj, prefix: str = "", separator: str = ".") -> Dict[str, str]:
```
Comment (MP91) [inline velocitas_lib/src/velocitas_lib/__init__.py]: could we move this to the variables? it is only used there

File: velocitas_lib/src/velocitas_lib/services.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+def parse_service_config(service_spec_config: Dict) -> ServiceSpecConfig:
+    """Parse service spec configuration and return it as an named tuple.
+
+    Args:
+        service_spec_config: The specificon of the services from config file.
+    """
+
+    is_enabled = True
+    no_dapr = False
```
Comment (MP91) [inline velocitas_lib/src/velocitas_lib/services.py]: I would prefer
```suggestion
    dapr = True
```
but this is not that important, just personal opinion

File: runtime-local/src/lib.py (hunk 1 above)
Comment (doosuu) [reply runtime-local/src/lib.py]: Good point!

File: velocitas_lib/src/velocitas_lib/__init__.py (hunk 2 above)
Comment (doosuu) [reply velocitas_lib/src/velocitas_lib/__init__.py]: Agreed

File: velocitas_lib/src/velocitas_lib/services.py (hunk 3 above)
Comment (doosuu) [reply velocitas_lib/src/velocitas_lib/services.py]: Agreed on `use_dapr`

File: velocitas_lib/src/velocitas_lib/services.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        open(
+            path,
+            encoding="utf-8",
+        )
+    )
+
+    services: List[Service] = list()
+    for service_json in json_array:
+        service_id = service_json["id"]
+        # service_interfaces = service_json['interfaces']
```
Comment (MP91) [inline velocitas_lib/src/velocitas_lib/services.py]: remove if not needed

File: velocitas_lib/src/velocitas_lib/services.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        service_config = ServiceSpecConfig(image="none")
```
Comment (MP91) [inline velocitas_lib/src/velocitas_lib/services.py]: Why setting this image to string none?
```suggestion
        service_config = ServiceSpecConfig()
```

**Review by MP91 — CHANGES_REQUESTED** [review]

File: velocitas_lib/src/velocitas_lib/services.py (hunk 5 above)
Comment (doosuu) [reply velocitas_lib/src/velocitas_lib/services.py]: Image is a mandatory parameter, if we set this to None, the typing would have to be `Optional[str]` but all of our current code relies on image to be present. But you are right, this should be handled differently.

**Review by MP91 — APPROVED** [review]
LGTM

**Pull request merged**